.github/workflows/release.yml

```diff
@@ -0,0 +1,30 @@
+name: Release
+
+on:
+  push:
+    branches: ["main"]
+
+jobs:
+  release:
+    name: Trigger release
+    runs-on: ubuntu-20.04
+    if: "!contains(github.event.head_commit.message, 'ci skip') && !contains(github.event.head_commit.message, 'skip ci')"
```

Comment:
As this don't introduce automatic version bump, I would use opt-in condition, rather than opt-out.